iodata.rb

```diff
@@ -0,0 +1,34 @@
+$LOAD_PATH << '.'
+require 'file_link'
+require 'csv'
+class IOData
+include File_link
+  def add_employee_data(emp_details_array)
+    File_link.write(emp_details_array[0],emp_details_array[1],emp_details_array[2])
+  end
+
+  def read_all_employee_data
+    csv = File_link.get_file_name
+    count = 0
+    puts "Displaying All Employee Data"
+    csv.each do |row| 
+      puts "#{row[0]} #{row[1]} #{row[2]}"
+      count += 1
+    end
+    puts "No of Employees: #{count}"
+  end
+
+  def show_dept_empl_data
+    csv = File_link.get_file_name
```

Comment:
repetition of code. DRY it up. 

You can have just 1 method to get data, a parameter to filter based on department or all departments. 

The actul puts statement should be outside the method. Only return the count and let the calling function use the puts method to decide whatever message it wants. 

Right now you cannot reuse these methods because its hard coded for emploee. Rather it should be more generic.